cloudinit/sources/DataSourceOVF.py

```diff
@@ -129,26 +129,82 @@ def _get_data(self):
                 else:
                     LOG.debug("Did not find the customization plugin.")
 
+                md_path = None
                 if vmwareImcConfigFilePath:
+                    imcdirpath = os.path.dirname(vmwareImcConfigFilePath)
+                    cf = ConfigFile(vmwareImcConfigFilePath)
+                    self._vmware_cust_conf = Config(cf)
                     LOG.debug("Found VMware Customization Config File at %s",
                               vmwareImcConfigFilePath)
-                    nicspath = wait_for_imc_cfg_file(
-                        filename="nics.txt", maxwait=10, naplen=5)
+                    try:
+                        (md_path, ud_path, nicspath) = collect_imc_files(
```

Comment:
let's rename this to collect_imc_file_paths so it's clearer what this function returns (returns paths instead of file contents)
```suggestion
                        (md_path, ud_path, nicspath) = collect_imc_file_paths(
```

Reply:
Thanks Chad for the detail comments, I addressed all of them. PTAL.